### PR TITLE
plugin/opensearchVersionCheck

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -1404,3 +1404,4 @@ module.exports = {
         'securityNotificationsEnabled'  : require(__dirname + '/plugins/alibaba/securitycenter/securityNotificationsEnabled.js')
     }
 };
+
```
Comment (alphadev4) [inline exports.js:1407]:
```suggestion
```

File: plugins/aws/opensearch/opensearchVersion.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'OpenSearch Version',
+    category: 'OpenSearch',
+    domain: 'Databases',
+    description: 'Checks if OpenSearch domains are running the latest  Engine version',
```
Comment (alphadev4) [inline plugins/aws/opensearch/opensearchVersion.js:8]:
```suggestion
    description: 'Checks if OpenSearch domains are running the latest engine version',
```

File: plugins/aws/opensearch/opensearchVersion.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    more_info: 'OpenSearch domains should be upgraded to the latest version for optimal performance and security.',
+    link: 'https://docs.aws.amazon.com/opensearch-service/latest/developerguide/features-by-version.html',
```
Comment (alphadev4) [inline plugins/aws/opensearch/opensearchVersion.js:10]:
```suggestion
    link: 'https://docs.aws.amazon.com/opensearch-service/latest/developerguide/what-is.html',
```

File: plugins/aws/opensearch/opensearchVersion.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    recommended_action: 'Ensure each OpenSearch domain is running the latest service software and update out-of-date domains.',
```
Comment (alphadev4) [inline plugins/aws/opensearch/opensearchVersion.js:11]:
```suggestion
    recommended_action: 'Update OpenSearch domain to set to latest engine version.',
```

File: plugins/aws/opensearch/opensearchVersion.js (hunk 3 above)
Comment (alphadev4) [reply plugins/aws/opensearch/opensearchVersion.js:10]: link should be relevant

File: plugins/aws/opensearch/opensearchVersion.js (hunk 4 above)
Comment (alphadev4) [reply plugins/aws/opensearch/opensearchVersion.js:11]: Recommended action should contain the recommended step to remediate.

File: plugins/aws/opensearch/opensearchVersion.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                return rcb();
+            }
+
+            if (!listDomainNames.data.length){
+                helpers.addResult(results, 0, 'No OpenSearch domains found', region);
+                return rcb();
+            }
+
+
+
```
[the remark this answers is not in the document]
Comment (alphadev4) [reply plugins/aws/opensearch/opensearchVersion.js:39]: avoid extra spaces

File: plugins/aws/opensearch/opensearchVersion.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                    !describeDomain.data ||
+                    !describeDomain.data.DomainStatus) {
+                    helpers.addResult(
+                        results, 3,
+                        'Unable to query for OpenSearch domain config: ' + helpers.addError(describeDomain), region);
+                } else {
+                    var localDomain = describeDomain.data.DomainStatus;
+                    var currentVersion = localDomain && localDomain.EngineVersion? localDomain.EngineVersion: '';
+
+                    if (currentVersion.includes('Elasticsearch')) {
```
Comment (mehakseedat63) [inline plugins/aws/opensearch/opensearchVersion.js:51]: This check can be simplified by doing something like
```suggestion
                    if ((currentVersion.includes('Elasticsearch') && currentVersion.includes('7.10')) || (currentVersion.includes('OpenSearch') && currentVersion.includes('2.5'))) {
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]
Fix linting issues

**Review by mehakseedat63 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**